txnsync/encodedgroupsunmarshalers.go

```diff
@@ -0,0 +1,644 @@
+	err = stub.BitmaskSender.iterate(int(stub.TotalTransactionsCount), len(stub.Sender)/crypto.DigestSize, func(i int, index int) error {
+		return nextSlice(&stub.Sender, signedTxns[i].Txn.Sender[:], crypto.DigestSize)
+	})
+	if err != nil {
+		return err
+	}
+	err = stub.BitmaskFee.iterate(int(stub.TotalTransactionsCount), len(stub.Fee), func(i int, index int) error {
+		signedTxns[i].Txn.Fee = stub.Fee[index]
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	err = stub.BitmaskFirstValid.iterate(int(stub.TotalTransactionsCount), len(stub.FirstValid), func(i int, index int) error {
+		signedTxns[i].Txn.FirstValid = stub.FirstValid[index]
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	err = stub.BitmaskLastValid.iterate(int(stub.TotalTransactionsCount), len(stub.LastValid), func(i int, index int) error {
+		signedTxns[i].Txn.LastValid = stub.LastValid[index]
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	err = stub.BitmaskNote.iterate(int(stub.TotalTransactionsCount), len(stub.Note), func(i int, index int) error {
+		signedTxns[i].Txn.Note = stub.Note[index]
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	err = stub.BitmaskGenesisID.iterate(int(stub.TotalTransactionsCount), int(stub.TotalTransactionsCount), func(i int, index int) error {
+		signedTxns[i].Txn.GenesisID = genesisID
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	for i := range signedTxns {
+		signedTxns[i].Txn.GenesisHash = genesisHash
+	}
+	err = stub.BitmaskLease.iterate(int(stub.TotalTransactionsCount), len(stub.Lease)/crypto.DigestSize, func(i int, index int) error {
+		return nextSlice(&stub.Lease, signedTxns[i].Txn.Lease[:], crypto.DigestSize)
+	})
+	if err != nil {
+		return err
+	}
+	err = stub.BitmaskRekeyTo.iterate(int(stub.TotalTransactionsCount), len(stub.RekeyTo)/crypto.DigestSize, func(i int, index int) error {
+		return nextSlice(&stub.RekeyTo, signedTxns[i].Txn.RekeyTo[:], crypto.DigestSize)
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (stub *txGroupsEncodingStub) reconstructKeyregTxnFields(signedTxns []transactions.SignedTxn) (err error) {
+	// should all have same number of elements
+	if len(stub.VotePK)/crypto.DigestSize != len(stub.VoteKeyDilution) || len(stub.SelectionPK)/crypto.DigestSize != len(stub.VoteKeyDilution) {
+		return errDataMissing
+	}
+	err = stub.BitmaskKeys.iterate(int(stub.TotalTransactionsCount), len(stub.VoteKeyDilution), func(i int, index int) error {
+		signedTxns[i].Txn.VoteKeyDilution = stub.VoteKeyDilution[index]
+		err := nextSlice(&stub.VotePK, signedTxns[i].Txn.VotePK[:], crypto.DigestSize)
+		if err != nil {
+			return err
+		}
+		return nextSlice(&stub.SelectionPK, signedTxns[i].Txn.SelectionPK[:], crypto.DigestSize)
+	})
+	if err != nil {
+		return err
+	}
+	err = stub.BitmaskVoteFirst.iterate(int(stub.TotalTransactionsCount), len(stub.VoteFirst), func(i int, index int) error {
+		if index >= len(stub.VoteFirst) {
```

Comment:
The index is already checked here. No need to check it inside iterate as well.
Similarly below. These are redundant checks.